crates/contract/src/call.rs

```diff
@@ -282,21 +281,20 @@ impl<P: TempProvider, D: CallDecoder> CallBuilder<P, D> {
         self
     }
 
-    /// Sets the `gas` field in the transaction to the provided value
-    pub fn gas(mut self, gas: U256) -> Self {
-        self.request = self.request.gas_limit(gas);
-        self
-    }
-
     /// Sets the `gas_price` field in the transaction to the provided value
     /// If the internal transaction is an EIP-1559 one, then it sets both
     /// `max_fee_per_gas` and `max_priority_fee_per_gas` to the same value
     pub fn gas_price(mut self, gas_price: U256) -> Self {
-        self.request = self.request.max_fee_per_gas(gas_price);
-        self.request = self.request.max_priority_fee_per_gas(gas_price);
+        match self.request.max_fee_per_gas {
+            None => self.request.gas_price = Some(gas_price),
+
+            Some(_) => {
+                self.request.max_fee_per_gas = Some(gas_price);
+                self.request.max_priority_fee_per_gas = Some(gas_price);
+            }
+        }
```

Comment:
with this change, we will never be able to set the EIP-1559 gas prices with this function, since `max_fee_per_gas` always starts out as `None`

Reply:
> with this change, we will never be able to set the EIP-1559 gas prices with this function, since `max_fee_per_gas` always starts out as `None`

thanks for your review, it should be max_fee_per_gas && max_priority_fee_per_gas ?

or i'm totally out of the context ?

Reply:
I'm not sure I follow, can you try to rephrase it?